### PR TITLE
Get rid of ALLOWED_NOT_READY env var

### PR DESCRIPTION
**Pull request by shyamjvs**
Now that https://github.com/kubernetes/kubernetes/pull/62639 is merged.
Actually, that change isn't cherrypicked on older releases - but I think it's fine to have ALLOWED_NOTREADY_NODES=0 for those jobs (as flakes from it are super rare).

/cc @wojtek-t

### CODE REVIEW

File: jobs/env/ci-kubernetes-e2e-gci-gce-scalability-beta.env [hunk 1]
```diff
@@ -1,7 +1,4 @@
 # Create a project k8s-e2e-gce-scalability-release and move this test there
 
-# Override GCE defaults.
```
[the remark this answers is not in the document]
Comment (shyamjvs) [reply jobs/env/ci-kubernetes-e2e-gci-gce-scalability-beta.env]: I know it isn't cherry-picked. See my PR description above :)

[the remark this answers is not in the document]
Comment (wojtek-t) [reply jobs/env/ci-kubernetes-e2e-gci-gce-scalability-beta.env]: Hmm - I'm not convinced it's fine, but let's try - we can revert those parts later.

**Pull request merged**